gems/smithy-client/lib/smithy-client/plugins/resolve_auth.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    module Plugins
+      # @api private
+      class ResolveAuth < Plugin
+        option(
+          :auth_resolver,
+          doc_default: '<DEFAULT_AUTH_RESOLVER>',
+          doc_type: '#resolve(context)',
+          rbs_type: 'Smithy::Client::AuthResolver',
+          docstring: 'An object that resolves authentication schemes for request signing'
+        )
+
+        # @api private
+        option(:auth_schemes) { {} }
+
+        def before_initialize(client_class, options)
+          options[:auth_resolver] ||= client_class.auth_resolver.new
+        end
+
+        # @api private
+        class Handler < Smithy::Client::Handler
+          def call(context)
+            # TODO: apply endpoint auth properties if present
+            auth_options = context.config.auth_resolver.resolve(context)
+            context.auth = resolve_auth(context, auth_options)
+            @handler.call(context)
+          end
+
+          private
+
+          def resolve_auth(context, auth_options)
+            failures = []
+
+            raise 'No auth options were resolved' if auth_options.empty?
+
+            auth_options.each do |auth_option|
+              identity_provider_config_option = context.config.auth_schemes[auth_option]
+
+              unless identity_provider_config_option
+                failures << "Auth scheme #{auth_option} was not enabled " \
```

Comment:
You can just make all of these a single line string - this was when rubocop enforced 80 character lines, it's now 120.